cdap-app-templates/cdap-etl/cdap-etl-data-pipeline/src/main/java/co/cask/cdap/etl/datapipeline/SmartWorkflow.java

```diff
@@ -172,8 +200,18 @@ private void addProgram(String phaseName, WorkflowProgramAdder programAdder) {
     }
     BatchPhaseSpec batchPhaseSpec = new BatchPhaseSpec(programName, phase, spec.getResources(),
                                                        spec.isStageLoggingEnabled(), phaseConnectorDatasets);
-    applicationConfigurer.addMapReduce(new ETLMapReduce(batchPhaseSpec));
-    programAdder.addMapReduce(programName);
+    switch (engine) {
+      case MAPREDUCE:
+        applicationConfigurer.addMapReduce(new ETLMapReduce(batchPhaseSpec));
+        programAdder.addMapReduce(programName);
+        break;
+      case SPARK:
+        applicationConfigurer.addSpark(new ETLSpark(batchPhaseSpec));
+        programAdder.addSpark(programName);
+        break;
+      default:
+        // should never happen
```

Comment:
would there be other engines we might want to support in the future?
in which case, it might be good to throw an exception in this case so its not silently happening.